Yosemite/YosemiteTests/Mockups/Networking/Remote/MockProductReviewsRemote.swift

```diff
@@ -0,0 +1,41 @@
+
+import Foundation
+import Networking
+
+import XCTest
+
+/// Mock for `ProductReviewsRemote`.
+///
+final class MockProductReviewsRemote: ProductReviewsEndpointsProviding {
+    private struct ResultKey: Hashable {
+        let siteID: Int64
+        let reviewID: Int64
+    }
+
+    /// The results to return based on the given arguments in `loadProductReview`
+    private var productReviewLoadingResults = [ResultKey: Result<ProductReview, Error>]()
+
+    /// Set the value passed to the `completion` block if `loadProductReview()` is called.
+    ///
+    func whenLoadingProductReview(siteID: Int64, reviewID: Int64, thenReturn result: Result<ProductReview, Error>) {
+        let key = ResultKey(siteID: siteID, reviewID: reviewID)
+        productReviewLoadingResults[key] = result
+    }
+
+    func loadProductReview(for siteID: Int64,
+                           reviewID: Int64,
+                           completion: @escaping (Result<ProductReview, Error>) -> Void) {
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else {
+                return
+            }
+
+            let key = ResultKey(siteID: siteID, reviewID: reviewID)
+            if let result = self.productReviewLoadingResults[key] {
+                completion(result)
+            } else {
+                XCTFail("\(String(describing: self)) Could not find Result for \(key)")
```

Comment:
nit: maybe this could just call `completion` with an error so that we could potentially use this remote mock to test network errors?

Reply:
(similar comment for `MockNotificationsRemote` and `MockProductsRemote`)

Reply:
I'm not sure about that. I did it that way on the first iteration. But it causes false-positives especially if the assertion is just:

```swift
XCTAssertTrue(result.isFailure)
```

You can still provide a network error simulation by:

```swift
whenLoadingProductReview(siteID: 1, reviewID: 1, thenReturn: .failure(.someError))
```

I think it's clearer if that's manually written in the unit test.